codegen/datamodelgen.py

```diff
@@ -162,7 +166,8 @@ def __init__(
             )
         if not self._static_info["solverworkflow"]:
             del self._static_info["solverworkflow"]
-        self._delete_generated_files()
+        if self.generate_py:
+            self._delete_generated_files()
```

Comment:
Please can we add a comment to clarify this logic. It can help future maintenance.

Reply:
Yes, will do.

Reply:
We are going to remove this code.